ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +92,217 @@ tests = testGroup "Mempool"
   , testProperty "removeTxs"                               prop_Mempool_removeTxs
```

Comment:
Surely some of them are redundant now that you have added the tests of the pure implementation? These should be removed.

Reply:
Since tests differs slightly in scope (either monadic or pure) I would keep all tests.